gateway-service/src/main/java/org/zowe/apiml/gateway/security/service/zosmf/AbstractZosmfService.java

```diff
@@ -77,15 +77,31 @@ protected String getZosmfServiceId() {
      */
     protected String getAuthenticationValue(Authentication authentication) {
         final String user = authentication.getPrincipal().toString();
-        final String password;
+        final char[] password;
         if (authentication.getCredentials() instanceof LoginRequest) {
             LoginRequest loginRequest = (LoginRequest) authentication.getCredentials();
             password = loginRequest.getPassword();
         } else {
-            password = (String) authentication.getCredentials();
+            password = readPassword(authentication.getCredentials());
```

Comment:
If credentials are stored as a string, there is a potential to create a new array without cleanup.

```
boolean cleanup = false;
```

```suggestion
            password = readPassword(authentication.getCredentials());
            cleanup = !(authentication.getCredentials() instanceof char[]);
```

```
try {
....
} finally {
  if (cleanup) {
    Arrays.fill(password, (byte) 0);
  }
}

Reply:
Feel free to perform the updates, we can have a look at the sync with v1